### ISSUE TITLE
Pb(100) The method getType() is undefined for the type

### ISSUE THREAD

**Issue by DaxiaK** [greenrobot/greenDAO#723]
Hi , 
I get a problem when I use gson & greenDAO at same time.
The code is :

```
@Entity
public class Foo implements IParse {

   @Id(autoincrement = false)
   private Long id;
    @ToMany
    @JoinEntity(
            entity = Foo2.class,
            sourceProperty = "id",
            targetProperty = "targetId"
    )
    private List<Foo2> foo2List;
//Others member
...
..
.
/**
 *  This is the IParse method
 */ 
 @Override
    public void parseSomeThing(JsonReader in) throws IOException {
        //Call next Name to route foo
        String Id = in.nextName();
        this.id = Long.valueOf(nextId);

        //parse the foo detail
        in.beginObject();
        while (in.hasNext()) {
            switch (in.nextName()) {
                case "A":
                   ...
                    break;
                case "B":
                    //Get error message here
                    this.foo2List = new Gson().fromJson(in,
                            new TypeToken<List<Foo2>>() {}
                                    .getType());
                    break;
                default:
                    //Skip the wrong json key
                    in.skipValue();
                    break;
            }
        }
        in.endObject();
    }
...
..
.
}
```

The error message is :
```
Error:Execution failed for task ':app:greendao'.
> Found 1 problem(s) parsing "MY/PROJECT/PATH". First problem:
  Pb(100) The method getType() is undefined for the type new TypeToken<List<Foo2>>(){} (67108964 at line 99).
  Run gradle with --info for more details.
```

Finally , I can fix it if I new the typetoken first:
```
TypeToken typeToken = new TypeToken<List<Foo2>>() {};
this.foo2List = new Gson().fromJson(in, typeToken.getType());
```

Do I miss something?
Thanks :)

**Comment by greenrobot-team** [comment]
This issue should have been fixed with version `3.2.2` of the Gradle plugin. Check if you have `classpath 'org.greenrobot:greendao-gradle-plugin:3.2.2'` in your build config. -ut

**Comment by DaxiaK** [comment]
I forget to update greendao-gradle-plugin.It work for me, thanks!
